paimon-vfs/paimon-vfs-hadoop/src/main/java/org/apache/paimon/vfs/hadoop/PaimonVirtualFileSystem.java

```diff
@@ -0,0 +1,418 @@
+        VFSIdentifier dstVfsIdentifier = vfsOperations.getVFSIdentifier(getVirtualPath(dst));
+        if (srcVfsIdentifier instanceof VFSCatalogIdentifier) {
+            throw new IOException("Cannot rename from virtual path " + src + " which is a catalog");
+        } else if (srcVfsIdentifier instanceof VFSDatabaseIdentifier) {
+            throw new IOException(
+                    "Cannot rename from virtual path " + src + " which is a database");
+        } else if (srcVfsIdentifier instanceof VFSTableRootIdentifier) {
+            throw new IOException("Cannot rename from virtual path " + src + " which is a table");
+        } else {
+            if (!(dstVfsIdentifier instanceof VFSTableIdentifier)) {
+                throw new IOException(
+                        "Cannot rename to virtual path " + dst + " which is not a table");
+            }
+            VFSTableIdentifier srcTableIdentifier = (VFSTableIdentifier) srcVfsIdentifier;
+            VFSTableIdentifier dstTableIdentifier = (VFSTableIdentifier) dstVfsIdentifier;
+            if (!srcTableIdentifier.isTableExist()) {
+                throw new IOException(
+                        "Cannot rename from virtual path "
+                                + src
+                                + " which is not in an existing table");
+            }
+            if (!dstTableIdentifier.isTableExist()) {
+                throw new IOException(
+                        "Cannot rename to virtual path "
+                                + dst
+                                + " which is not in an existing table");
+            }
+            TableMetadata srcTable = srcTableIdentifier.getTable();
+            TableMetadata dstTable = dstTableIdentifier.getTable();
+            if (!srcTable.uuid().equals(dstTable.uuid())) {
+                throw new IOException(
+                        "Cannot rename from virtual path "
+                                + src
+                                + " to virtual path "
+                                + dst
+                                + " which is not in the same table");
+            }
+            return srcTableIdentifier
+                    .fileIO()
+                    .rename(srcTableIdentifier.getRealPath(), dstTableIdentifier.getRealPath());
+        }
+    }
+
+    @Override
+    public boolean delete(Path f, boolean recursive) throws IOException {
+        VFSIdentifier vfsIdentifier = vfsOperations.getVFSIdentifier(getVirtualPath(f));
+        if (vfsIdentifier instanceof VFSCatalogIdentifier) {
+            throw new IOException("Cannot delete virtual path " + f + " which is a catalog");
+        } else if (vfsIdentifier instanceof VFSDatabaseIdentifier) {
+            throw new IOException("Cannot delete virtual path " + f + " which is a database");
+        } else if (vfsIdentifier instanceof VFSTableRootIdentifier) {
+            throw new IOException("Cannot delete virtual path " + f + " which is a table");
```

Comment:
We can support delete table.

Reply:
1. Can we support delete database in pvfs

Reply:
I think yes.